src/act_info.cpp

```diff
@@ -2009,19 +1990,15 @@ void do_credits(CHAR_DATA *ch, const char *argument) {
 void do_where(CHAR_DATA *ch, const char *argument) {
     char buf[MAX_STRING_LENGTH];
     char arg[MAX_INPUT_LENGTH];
-    CHAR_DATA *victim;
-    Descriptor *d;
-    bool found;
 
     one_argument(argument, arg);
 
     if (arg[0] == '\0') {
-        snprintf(buf, sizeof(buf), "|cYou are in %s\n\rPlayers near you:|w\n\r", ch->in_room->area->areaname);
-        send_to_char(buf, ch);
-        found = false;
-        for (d = descriptor_list; d; d = d->next) {
-            if (d->is_playing() && (victim = d->character()) != nullptr && !IS_NPC(victim) && victim != ch
-                && victim->in_room != nullptr && victim->in_room->area == ch->in_room->area && can_see(ch, victim)) {
+        send_to_char("|cYou are in {}\n\rPlayers near you:|w\n\r"_format(ch->in_room->area->areaname), ch);
+        auto found = false;
+        for (auto &d : descriptors().all_visible_to(ch)) {
```

Comment:
If we find that there's more scenarios where we need to only collate the ch who are in the same area, this can probably become a descriptor streamer in its own right.

Reply:
Right. This is a good argument for the chainable filters: I thought a couple of times about "same room as" and "same zone as", but some were "including ch" and some "excluding ch", which I couldn't phrase with my monolithic functions. Definitely worth a think to have `all_in_zone()` and `all_in_room()` with a `| excluding(ch)` going on ... hmm. thanks!